htdocs/web_portal/controllers/downtime/view_endpoint_tree.php

```diff
@@ -19,19 +19,30 @@
  * limitations under the License.
  *
  /*====================================================== */
+use Doctrine\Common\Collections\ArrayCollection;
+use Exception;
+
 function getServiceandEndpointList() {
     require_once __DIR__ . '/../utils.php';
     require_once __DIR__ . '/../../../web_portal/components/Get_User_Principle.php';
 
+    $params = [];
     $dn = Get_User_Principle();
     $user = \Factory::getUserService()->getUserByPrinciple($dn);
     $params['portalIsReadOnly'] = portalIsReadOnlyAndUserIsNotAdmin($user);
+    $siteIDs = $_REQUEST['site_id'];
 
-    if (!isset($_REQUEST['site_id']) || !is_numeric($_REQUEST['site_id']) ){
-        throw new Exception("An id must be specified");
+    if (empty($siteIDs)) {
```

Comment:
is this change equivalent to the old functionality? what if a non numeric string is passed in via `site_id`?

Reply:
when you loop through the siteIDs later, can you add an `is_numeric` check?